### PR TITLE
Improved rendering of drop shadow for Qt

### PR DESCRIPTION
**Pull request by tronical**
Use QGraphicsBlurEffect as public API to take the blur into account

### CODE REVIEW

**Review by ogoffart — APPROVED** [review]

File: sixtyfps_runtime/rendering_backends/qt/qt_window.rs [hunk 1]
```diff
@@ -532,19 +542,89 @@ impl ItemRenderer for QtItemRenderer<'_> {
...
+
+                        // takes ownership of the effect
+                        pixmap_item->setGraphicsEffect(blur_effect);
+
+                        QRectF scene_rect(-blur_radius, -blur_radius, img->width() + 2. * blur_radius, img->height() + 2. * blur_radius);
+                        QImage blurred_scene(scene_rect.size().toSize(), QImage::Format_ARGB32_Premultiplied);
+                        blurred_scene.fill(Qt::transparent);
+
+                        QPainter p(&blurred_scene);
+                        scene.render(&p, QRectF(), scene_rect);
```
[the remark this answers is not in the document]
Comment (tronical) [reply sixtyfps_runtime/rendering_backends/qt/qt_window.rs:595]: Good idea, I'll give it a try. `setGraphicsEffect` should register the item as the source for the effect and then it should work.

File: sixtyfps_runtime/rendering_backends/qt/qt_window.rs [hunk 2]
```diff
@@ -176,6 +176,13 @@ cpp! {{
         }
     };
 
+    class SixtyFPSPublicGraphicsBlurEffect : public QGraphicsBlurEffect {
```
Comment (ogoffart) [inline sixtyfps_runtime/rendering_backends/qt/qt_window.rs:179]: (I would just have put this class in the function that uses it that in the function which uses it.)

File: sixtyfps_runtime/rendering_backends/qt/qt_window.rs [hunk 3]
```diff
@@ -176,6 +176,13 @@ cpp! {{
...
+    public:
+        // Make public what's protected
+        void draw_public(QPainter *p) {
```
Comment (ogoffart) [inline sixtyfps_runtime/rendering_backends/qt/qt_window.rs:182]: `using QGraphicsBlurEffect::draw;`

**Review by ogoffart — APPROVED** [review]

File: sixtyfps_runtime/rendering_backends/qt/qt_window.rs (hunk 3 above)
Comment (tronical) [reply sixtyfps_runtime/rendering_backends/qt/qt_window.rs]: Makes perfect sense :). Thanks, I'm too old fashioned sometimes.

Comment (ogoffart) [reply sixtyfps_runtime/rendering_backends/qt/qt_window.rs]: That's good old C++98

**Review by ogoffart — APPROVED** [review]

**Pull request merged**